File: java/InsecureDeserialization.java
import java.io.FileInputStream;
import java.io.ObjectInputStream;

public class InsecureDeserialization {
    public static void main(String[] args) throws Exception {
        // CWE-502: deserializing untrusted data from a file allows arbitrary code execution.
        ObjectInputStream ois = new ObjectInputStream(new FileInputStream(args[0]));
        ois.setObjectInputFilter(ObjectInputFilter.Config.createFilter("java.lang.String;!*"));
        Object obj = ois.readObject();
        System.out.println(obj);
        ois.close();
    }
}

File: csharp/InsecureDeserialization.cs
using System;
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;

namespace SastSamples
{
    public static class InsecureDeserialization
    {
        public static void Main(string[] args)
        {
            // CWE-502: BinaryFormatter deserializes untrusted data allowing remote code execution.
#pragma warning disable SYSLIB0011
            var formatter = new BinaryFormatter();
#pragma warning restore SYSLIB0011
            using var stream = new FileStream(args[0], FileMode.Open);
            var obj = formatter.Deserialize(stream);
            Console.WriteLine(obj);
        }
    }
}
